azext_edge/edge/providers/orchestration/targets.py

```diff
@@ -259,12 +261,32 @@ def get_broker_config_target_map(self):
 
         return processed_config_map
 
-    def get_advanced_config_target_map(self):
+    def get_advanced_config_target_map(self) -> dict:
         processed_config_map = {}
         if self.enable_fault_tolerance:
             processed_config_map["edgeStorageAccelerator"] = {"faultToleranceEnabled": True}
 
-        return {"advancedConfig": processed_config_map}
+        return processed_config_map
+
+    def get_trust_settings_target_map(self) -> dict:
+        source = "SelfSigned"
+        result = {"source": source}
+        if self.trust_settings:
+            target_settings: Dict[str, str] = {}
+            result["source"] = "CustomerManaged"
+            trust_bundle_def = M2_ENABLEMENT_TEMPLATE.get_type_definition("_1.TrustBundleSettings")["properties"]
+            allowed_issuer_kinds: Optional[List[str]] = trust_bundle_def.get("issuerKind", {}).get("allowedValues")
+            for key in TRUST_SETTING_KEYS:
+                if key not in self.trust_settings:
+                    raise InvalidArgumentValueError(f"{key} is a required trust setting/key.")
+                if key == TRUST_ISSUER_KIND_KEY:
+                    if allowed_issuer_kinds:
```

Comment:
`if allowed_issuer_kinds and self.trust_settings[key] not in allowed_issuer_kinds:` ?